### PR TITLE
Angular update

### PR DESCRIPTION
**Pull request by dacowan404**

### CODE REVIEW

**Comment by billswitzer2** [comment]
Thanks David! This is awesome. Bill

From: David Cowan ***@***.***>
Sent: Wednesday, April 10, 2024 11:45 AM
To: CDCgov/MicrobeTrace ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [CDCgov/MicrobeTrace] Angular update (PR #796)

CAUTION: This email originated from outside of the organization. Do not click links or open attachments unless you recognize the sender and know the content is safe.
________________________________

________________________________
You can view, comment on, or merge this pull request online at:

  https://github.com/CDCgov/MicrobeTrace/pull/796

Commit Summary

  *   703de25<https://github.com/CDCgov/MicrobeTrace/pull/796/commits/703de25dd5a3a2dad995ae3c531c3399267df0e8> updated alignment view (changed how seq are stored which let me add ability to change labels, sort based on different fields, and search with any field
  *   23e8f61<https://github.com/CDCgov/MicrobeTrace/pull/796/commits/23e8f61a96097fb07c13056044afd3b69ec316fe> added export visualization and export data functionalities to alignment view; also limits height of minimap to 100px; in files view users can generate random seqs, view preview alignment, and fixed bug to allow users to use fasta files again
  *   d951e5d<https://github.com/CDCgov/MicrobeTrace/pull/796/commits/d951e5d0d0497b4fa2ab8c85f550240616d9bae9> in alignment view, converted all the required variables to widgets to allow users to save settings between sessions
  *   1efafab<https://github.com/CDCgov/MicrobeTrace/pull/796/commits/1efafabaf914c7d6d9954bd2857dd8512f8d80a1> fixed a bug in alignment view where updateMiniMap was called before it was ready and updateAlignment was called twice in ngOnInit
  *   e0c8dd9<https://github.com/CDCgov/MicrobeTrace/pull/796/commits/e0c8dd9eb17a7f857ef76f5f4d2fe94b1970fc82> style changes to alignment view (text for titles are smaller on alignment view, added more description to barplot/logo/ruler)
  *   f40311f<https://github.com/CDCgov/MicrobeTrace/pull/796/commits/f40311fd3f501ebbac891354145e6354b49db310> Merge branch 'angularUpdate' of https://github.com/mossy426/MicrobeTrace into angularUpdate
  *   0ab78d3<https://github.com/CDCgov/MicrobeTrace/pull/796/commits/0ab78d36f4aa1375b061cb7756da94586fc12110> Update inline worker to dynamically use latest patristic and tn93 code
  *   971ae7c<https://github.com/CDCgov/MicrobeTrace/pull/796/commits/971ae7cc91a706d04cde3eaa6b0454376e98a3a7> fixed bugs relating to sizing in Table View (now has horizonal and vertical scroll bars, pagination (change page, number of rows/page) is now centered on screen, added ability to show all rows on a single page). also added JSDocs to most of table component
  *   02d4129<https://github.com/CDCgov/MicrobeTrace/pull/796/commits/02d41299e5b773a27c959edf1a83c675615c0e3d> Merge branch 'angularUpdate' of https://github.com/mossy426/MicrobeTrace into angularUpdate
  *   fb2f535<https://github.com/CDCgov/MicrobeTrace/pull/796/commits/fb2f535865e54a84bc5482531b2c85dfef5ac152> fixed table view export; now can export all columns or selected/visible columns as either csv or excel file
  *   35a34b0<https://github.com/CDCgov/MicrobeTrace/pull/796/commits/35a34b06044d8ddcee870903bfa89dc31b309cb5> allows importing and exporting of style files
  *   4492360<https://github.com/CDCgov/MicrobeTrace/pull/796/commits/44923608ee27ad710d93b0b51a12c0150a7180f6> table component (resizing row height/font size, can change order of columns, fixed filtering applying to other views)
  *   6d059a3<https://github.com/CDCgov/MicrobeTrace/pull/796/commits/6d059a3cfbdd7822893949166370de1dfb7c2bed> Fixed some bugs for tiling mulitple views in a dashboard (specifically alignment, table, map views), updated table settings format, fixed bug in map view where node positions would reroll too frequently (such as on node transparency or basemap changes) and link positions wouldn't alway be synced, fixed bug with files readFastas function
  *   4de68d5<https://github.com/CDCgov/MicrobeTrace/pull/796/commits/4de68d5acfad7c16518066101032d3a5e7976b22> Updated phylogenic tree and epi view to resize appropriately in tiling/dashboard view. Also updated settings and export dialog boxes to display on top and not behind other views
  *   f14adc6<https://github.com/CDCgov/MicrobeTrace/pull/796/commits/f14adc698db1f64e2dc2757126ff4017248d1241> fixed a bug from previous commit where settings and export dialog boxes wouldn't disappear when the view become unactive
  *   b9f91e0<https://github.com/CDCgov/MicrobeTrace/pull/796/commits/b9f91e0bd372ba77e6df7d0f7216a8add36ff6f1> Fixed tooltip positioning when multiple views are visible for alignment, map, and tree view; fixed map reroll jitter button; map view, node border affected by transparency setting; 2d setting for node and link tooltip are no longer displayed underneath settings box

File Changes

(27 files<https://github.com/CDCgov/MicrobeTrace/pull/796/files>)

  *   M angular.json<https://github.com/CDCgov/MicrobeTrace/pull/796/files#diff-2d2675bb4687601a5c7ccf707455132f90f3516a33716185687e5c41df59ac7d> (5)
  *   M package-lock.json<https://github.com/CDCgov/MicrobeTrace/pull/796/files#diff-053150b640a7ce75eff69d1a22cae7f0f94ad64ce9a855db544dda0929316519> (28)
  *   M package.json<https://github.com/CDCgov/MicrobeTrace/pull/796/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519> (4)
  *   M src/app/contactTraceCommonServices/common.service.ts<https://github.com/CDCgov/MicrobeTrace/pull/796/files#diff-2e7e3ba720b74c03a45e250d7cab77ab0cc6d598c8ca9dabb0c68953a8a1a1f2> (42)
  *   M src/app/filesComponent/files-plugin.component.html<https://github.com/CDCgov/MicrobeTrace/pull/796/files#diff-c16e1d77f81fb70f2ef7e06730435f81bfa8e76757b1e2af952253046491947b> (14)
  *   M src/app/filesComponent/files-plugin.component.ts<https://github.com/CDCgov/MicrobeTrace/pull/796/files#diff-ad6e41b21a8f0788c93f7efbd5a4ff399e93d0cdcb9462374dbedd5a11d1a765> (144)
  *   M src/app/helperClasses/inlineWorker.ts<https://github.com/CDCgov/MicrobeTrace/pull/796/files#diff-8a622a71c867884390e9b27cd41d8fe8d59bf32cdf8a25ed09e4efc2bb4e7ae8> (28)
  *   M src/app/helperClasses/interfaces.ts<https://github.com/CDCgov/MicrobeTrace/pull/796/files#diff-52024b311d340e56869c1fac5b338e9a31edb27e209eb116b0da62f446a678f8> (4)
  *   M src/app/microbe-trace-next-plugin.component.html<https://github.com/CDCgov/MicrobeTrace/pull/796/files#diff-93eaf186c174343c986ba14022c42a82e7cc4546f979fb27f13a24ae84270b5a> (13)
  *   M src/app/microbe-trace-next-plugin.component.ts<https://github.com/CDCgov/MicrobeTrace/pull/796/files#diff-6ad08f6cbfdfca0345e0c32b01b4729de4a3d8c88d6f900445a787b2bbcfa68f> (63)
  *   M src/app/visualizationComponents/AlignmentViewComponent/alignment-view-plugin-component.html<https://github.com/CDCgov/MicrobeTrace/pull/796/files#diff-c353fe35fea1007f5483bf62f549cca2a234b9e7cac08a11860be5c270391e46> (211)
  *   M src/app/visualizationComponents/AlignmentViewComponent/alignment-view-plugin-component.scss<https://github.com/CDCgov/MicrobeTrace/pull/796/files#diff-5239c1aaed252677091319df3b7537f79c1839dfe7594a406c885577a331725c> (2)
  *   M src/app/visualizationComponents/AlignmentViewComponent/alignment-view-plugin-component.ts<https://github.com/CDCgov/MicrobeTrace/pull/796/files#diff-a7e555e7ef7c5a809cfc0f32de0cd94f2f9a42cf18fbe83367446a4c575178a7> (610)
  *   M src/app/visualizationComponents/AlignmentViewComponent/generateAlignmentViewCanvas.ts<https://github.com/CDCgov/MicrobeTrace/pull/796/files#diff-f2249a99869887a0b62905dad8ac78784e9fe0c1f95f027afc5b89e4a5aa15f4> (15)
  *   M src/app/visualizationComponents/MapComponent/map-plugin.component.html<https://github.com/CDCgov/MicrobeTrace/pull/796/files#diff-a953be60d43ae4c4a5796e2e6ac51b5dec7197d41d476cb45341f9bad2e1dbc3> (12)
  *   M src/app/visualizationComponents/MapComponent/map-plugin.component.ts<https://github.com/CDCgov/MicrobeTrace/pull/796/files#diff-11d9f01c695875fb76d943b15ed87af45e40313b6fad4cd7436483a5001b69ed> (151)
  *   M src/app/visualizationComponents/PhylogeneticComponent/phylogenetic-plugin.component.html<https://github.com/CDCgov/MicrobeTrace/pull/796/files#diff-75d4021374f5a144497f35a034a65aee96ce092461818c3411a73a571ac4a102> (8)
  *   M src/app/visualizationComponents/PhylogeneticComponent/phylogenetic-plugin.component.ts<https://github.com/CDCgov/MicrobeTrace/pull/796/files#diff-73031da93e12fe18713ed1c0e2c30a4be1d8a036e1b00e767f0fb80776273f16> (33)
  *   M src/app/visualizationComponents/TableComponent/table-plugin-component.html<https://github.com/CDCgov/MicrobeTrace/pull/796/files#diff-0f3a1cb38cd5b294fa3c5907c9ca72ee69262b0658696d98a89a933dadc122aa> (105)
  *   M src/app/visualizationComponents/TableComponent/table-plugin-component.less<https://github.com/CDCgov/MicrobeTrace/pull/796/files#diff-4c24baa5daaa82f95f32bd0f77a27805621a02b73aaa29ec4ddc20a8918145ae> (7)
  *   M src/app/visualizationComponents/TableComponent/table-plugin-component.ts<https://github.com/CDCgov/MicrobeTrace/pull/796/files#diff-0b78a2b602a0d4473ca156548317762f8165e2e9c55f26146ccf300e581be86b> (205)
  *   M src/app/visualizationComponents/TimelineComponent/timeline-component.component.html<https://github.com/CDCgov/MicrobeTrace/pull/796/files#diff-52facab846ac42fbc4fee4a3a365918078cb1a070b7aeb89614e6201707fce70> (4)
  *   M src/app/visualizationComponents/TimelineComponent/timeline-component.component.ts<https://github.com/CDCgov/MicrobeTrace/pull/796/files#diff-ac189c72b391bbf0c27fad862f3b1de7094b3301ef353e7422b34cac0ac32209> (29)
  *   M src/app/visualizationComponents/TwoDComponent/twoD-plugin.component.html<https://github.com/CDCgov/MicrobeTrace/pull/796/files#diff-22da4de50d60ca03b7dd36ef305d280679316828607e21186cda258c941761c6> (21)
  *   M src/app/visualizationComponents/TwoDComponent/twoD-plugin.component.ts<https://github.com/CDCgov/MicrobeTrace/pull/796/files#diff-a07de1e0e55ecc7e95805f890501d16a509acba1e7edb39fef69f979d67e4f56> (26)
  *   M src/app/workers/workModule.ts<https://github.com/CDCgov/MicrobeTrace/pull/796/files#diff-d89d9eadca5fdd67d6cf7fbf4ac8138f6dfa346deeb03c48570d9b96d9b5e5bd> (2303)
  *   M src/styles.scss<https://github.com/CDCgov/MicrobeTrace/pull/796/files#diff-23fad3928bf3e71585eb4a6e3b2ff72dae02c3ed5a44d2f0deb1d49f0cb3b1a3> (16)

Patch Links:

  *   https://github.com/CDCgov/MicrobeTrace/pull/796.patch
  *   https://github.com/CDCgov/MicrobeTrace/pull/796.diff

File: angular.json
Comment (mossy426-cdc) [inline angular.json:1]: @dacowan404  Curious why you added this?

File: src/app/filesComponent/files-plugin.component.html [hunk 1]
```diff
@@ -41,14 +41,14 @@ <h1><b>Please add data files to load...</b></h1>
                                             </select>
                                         </div>
                                     </div>
-                                    <div id="ambiguities-row" class="form-group row">
+                                    <div id="ambiguities-row" *ngIf="SelectedDefaultDistanceMetricVariable=='tn93'" class="form-group row">
```
Comment (mossy426-cdc) [inline src/app/filesComponent/files-plugin.component.html:44]: Was this an element that slide up/down to show/hide?  If it was we should stick to the slide up/down animation.  If not, then you can ignore this.

File: src/styles.scss [hunk 2]
```diff
@@ -562,4 +562,20 @@ body .ui-tabview.ui-tabview-top .ui-tabview-nav li.ui-state-active, body .ui-tab
 
 .ui-table-scrollable-body {
     max-height: 665px;
+}
+
+body {
```
Comment (mossy426-cdc) [inline src/styles.scss:567]: What was this styling for in the accordion?

Comment (dacowan404) [reply src/styles.scss:567]: yes. In order to get the settings menus to not display behind different views in when tiling different views, I appended the dialog boxes to body and therefore had to update the css.

File: angular.json
Comment (dacowan404) [reply angular.json:1]: Looks like this is from a commit from @reagank

File: src/app/filesComponent/files-plugin.component.html (hunk 1 above)
Comment (dacowan404) [reply src/app/filesComponent/files-plugin.component.html:44]: The current deployed dev site slides up/down which does look better than mossy site. I think I was trying to get the file settings and global setting distance metrics to sync but forgot to revisit it and finish fixing the bug. Currently on both dev and mossy, updates of distance metric and link threshold variables on file setting updates global settings but updates of these variable on global settings doesn't update file setting. Please let me know how I should resolve this.

Comment (mossy426-cdc) [reply src/app/filesComponent/files-plugin.component.html:44]: Got it, yes updates should be 2-way.

**Pull request merged**